### PR TITLE
Tezos DataTypes

### PR DESCRIPTION
**Pull request by rlkelly**
## Description

This adds the necessary components for Tezos development.

## Testing instructions

Tests:
    - swift/Tests/Blockchains/TezosTests.swift
    - swift/Tests/Addresses/TezosAddressTests.swift
    - tests/Tezos/*

## Types of changes

Introduces types for building Tezos transactions, and methods to forge these transactions to the proper format, along with native interfaces.

## Checklist

- [ x ] Prefix PR title with `[WIP]` if necessary.
- [ x ] Add tests to cover changes as needed.
- [ x ] Update documentation as needed.

### CODE REVIEW

**Comment by keefertaylor** [comment]
Merging changes...

**Comment by rlkelly** [comment]
@alejandro-isaza any chance we can get this merged, it's pretty unwieldy.  I can make a separate PR for the C classes

**Comment by alejandro-isaza** [comment]
Looks like this PR adds back the google test submodule, please remove that.

**Comment by alejandro-isaza** [comment]
Also it needs to be merged with the latest master.

**Comment by alejandro-isaza** [comment]
Also tests failed:
```Undefined symbols for architecture x86_64:
  "forgePublicKeyHash(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      TW::Tezos::Address::forge() const in Address-439290af3b3b28988b6b10d275303907d4df479b74194097ad05ddea1dcb2539.o
  "bytesToBase58(unsigned char const*, unsigned long)", referenced from:
      TW::Tezos::Address::string() const in Address-439290af3b3b28988b6b10d275303907d4df479b74194097ad05ddea1dcb2539.o
  "base58CheckDecodePrefix(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned long, unsigned char*, unsigned char*)", referenced from:
      TW::Tezos::Address::Address(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in Address-439290af3b3b28988b6b10d275303907d4df479b74194097ad05ddea1dcb2539.o
      TW::Tezos::Address::forge() const in Address-439290af3b3b28988b6b10d275303907d4df479b74194097ad05ddea1dcb2539.o
ld: symbol(s) not found for architecture x86_64```

**Comment by alejandro-isaza** [comment]
I can help fixing the Xcode project (where the error is coming from) if you need me to.

File: tests/interface/TWHDWalletTests.cpp [hunk 1]
```diff
@@ -91,10 +91,11 @@ TEST(HDWallet, DeriveBitcoin) {
 TEST(HDWallet, DeriveTezos) {
     auto wallet = WRAP(TWHDWallet, TWHDWalletCreateWithMnemonic(words.get(), passphrase.get()));
     auto key = WRAP(TWPrivateKey, TWHDWalletGetKey(wallet.get(), TWCoinTypeTezos, 0, 0, 0));
-    auto publicKey = TWPrivateKeyGetPublicKeySecp256k1(key.get(), false);
+    auto publicKey = TWPrivateKeyGetPublicKeyEd25519(key.get());
     auto publicKeyData = WRAPD(TWPublicKeyData(publicKey));
```
Comment (vikmeup) [inline tests/interface/TWHDWalletTests.cpp]: Tezos should be using `Secp256k1` ?

Comment (rlkelly) [reply tests/interface/TWHDWalletTests.cpp]: this test is failing, I can remove these changes until i get it working right.

Comment (keefertaylor) [reply tests/interface/TWHDWalletTests.cpp]: I am going to revert this file.

If there is a bug in how we handle addresses, lets address in a separate PR and try to keep this PR minimally focused.

I'll follow up later today about proper address derivation.

**Pull request merged**